files

lopesnuno opened the pull request.

lopesnuno committed:
files

kimile commented on db.ddl.sql:
```diff
@@ -0,0 +1,52 @@
+CREATE TABLE room
+(
+    id         varchar NOT NULL,
```
Porque é que o varchar não tem length definida? Se é de 36 (UUID) deverias definir o valor e, nessa caso, nem usar varchar mas sim char(36)... 
Como vamos usar um random aplicacional, coloca `char(20)`

kimile replied:
Coloquei um ficheiro chamado random.js que gera um ID... `node random.js` (se não funcionar, deita a máquina ao lixo)

kimile commented on db.ddl.sql:
```diff
@@ -0,0 +1,52 @@
+    CONSTRAINT pk_account PRIMARY KEY (id)
+);
+
+CREATE TABLE room_meeters
```
`room_members`

kimile commented on db.ddl.sql:
```diff
@@ -0,0 +1,52 @@
+    CONSTRAINT pk_message PRIMARY KEY (id)
+);
+
+ALTER TABLE message
+    ADD sender_id    varchar NOT NULL
+        CONSTRAINT fk_message_sender REFERENCES account (id),
+    ADD recipient_id varchar
+        CONSTRAINT fk_message_recipient REFERENCES account (id),
+    ADD reply_to     varchar
+        CONSTRAINT fk_message_reply REFERENCES message (id),
+    ADD room_id      varchar
+        CONSTRAINT fk_message_room REFERENCES room (id);
```
Porque é que estas intruções não foram directamente declaradas no create table?

kimile commented on fixtures.sql:
```diff
@@ -0,0 +1,37 @@
+        'a945be2b-bad6-48ae-831c-04cc809cf189', NULL, NULL);
+
+INSERT INTO room(id, name, created_at, updated_at, owner)
+VALUES (gen_random_uuid(), 'Bilcas e borgas', 1423759234, 1486917634, 'a945be2b-bad6-48ae-831c-04cc809cf189');
```
Corrige o `bilcas`

kimile commented on fixtures.sql:
```diff
@@ -0,0 +1,37 @@
+INSERT INTO account(id, name, created_at, updated_at)
+VALUES (gen_random_uuid(), 'Joaquim', 1392223234, 1615567234);
```
Usa o random.js para gerar ids

kimile commented on db.ddl.sql:
```diff
@@ -0,0 +1,52 @@
+    id         varchar NOT NULL,
+    name       varchar(50),
+    created_at numeric,
+    updated_at numeric,
+    constraint pk_room PRIMARY KEY (id)
```
Falta o owner (fk para account)

lopesnuno committed:
Fixes

Fixed constraints;
Change ID types and "room_meeters" to room_members.

kimile commented on db.ddl.sql:
```diff
@@ -0,0 +1,46 @@
+    constraint pk_room PRIMARY KEY (id)
+);
+
+CREATE TABLE room_members
```
Não é uma regra, mas é uma prática em muitos projectos de usar os nomes das tabelas no singular! Apesar de não termos definido, como todas as outras tabelas têm nomes no singular, vamos colocar esta também (`room_member`)

kimile replied:
Pensando bem, vou reverter o comentário anterior... a regra vai ser usar o plural... nesse sentido:
`account` -> `users`
`room` -> `rooms`
`room_members` -> `room_members`
`message` -> `messages`

kimile commented on db.ddl.sql:
```diff
@@ -0,0 +1,46 @@
+    updated_at numeric,
+    owner      char(20) NOT NULL
+        CONSTRAINT fk_room_owner REFERENCES account (id),
+    constraint pk_room PRIMARY KEY (id)
```
Tens todos os constraints em caps (como deve ser), coloca este também!!

kimile reviewed (changes requested):
Falta pouco!

kimile commented on db.ddl.sql:
```diff
@@ -0,0 +1,46 @@
+(
+    id         char(20) NOT NULL,
+    name       varchar(50),
+    created_at numeric  NOT NULL,
+    updated_at numeric,
```
Tal como te tinha dito, é melhor para a tua experiência usares o tipo "tradicional" aqui, pelo que vais mudar os `created_at` e `updated_at` para o tipo `TIMESTAMP` em todas as tabelas!
Assim treinas queries com date types.

lopesnuno committed:
Fixes_2

- Changed table names do plurar,
- constraint -> CONSTRAINT,
- Dates to "TIMESTANT".

lopesnuno committed:
use timestamp and formatting

send_at and joined_at to TIMESTAMP
